osu.Game/Beatmaps/BeatmapManager.cs

```diff
@@ -101,12 +100,20 @@ public WorkingBeatmap CreateNew(RulesetInfo ruleset, User user)
         /// <summary>
         /// Fired when a single difficulty has been hidden.
         /// </summary>
-        public IBindable<WeakReference<BeatmapInfo>> BeatmapHidden => beatmapModelManager.BeatmapHidden;
+        public Action<BeatmapInfo> BeatmapHidden
+        {
+            get => beatmapModelManager.BeatmapHidden;
+            set => beatmapModelManager.BeatmapHidden = value;
+        }
```

Comment:
I don't really follow this structure. Why is this not declared as an event, and why is the delegate freely externally mutable rather than just allowing add/remove as events do? Also not following why the delegated `BeatmapModelManager` delegates are not events and also are externally mutable.

Same question goes everywhere else, because it looks like all changes to these events/callbacks are done in the same manner elsewhere in this diff.

Reply:
all temporary but will change

Reply:
I see a new commit here that changed `Item{Updated,Removed}` and `Download{Began,Failed}` to use events, but `Beatmap{Hidden,Restored}` look to be unchanged, so unsure if omitted by accident or on purpose.

Reply:
Neither accidental or on purpose, just planned for a follow up commit. Should happen today unless you beat me to it.